frontend/src/metabase/visualizations/echarts/cartesian/option/axis.ts

```diff
@@ -341,7 +341,7 @@ export const buildCategoricalDimensionAxis = (
       ...getHistogramTicksOptions(chartModel, settings, chartMeasurements),
       formatter: (value: string) => {
         const numberValue = parseNumberValue(value);
-        if (isNumber(column) && numberValue !== null) {
+        if (isNumericBaseType(column) && numberValue !== null) {
```

Comment:
@alxnddr I do not know if this is the right change to make. 

This all seems to be stemming from the fact that when we are using the `"ordinal"` scale, ECharts is returning the data values as strings instead of their original type - in this case numbers:

**Dataset**
![image](https://github.com/metabase/metabase/assets/22608765/f7fb9a10-8de5-4bcf-acf5-55ab3f2b283c)

Because it is returning them as strings, we have to parse if it is a number or not, otherwise we will not get the correct formatting applied based on the column settings.

So there are two options:
1. Get ECharts to call the formatter with unaltered types and not worry about correctly parsing the value from a string
   - i.e. do not convert them to strings
2. Check the column type and if the column is numeric, parse it to a number and use that number in the formatter

But I do not think it is currently possible to get ECharts to return with unaltered types, at least according to the types they have: 

![image](https://github.com/metabase/metabase/assets/22608765/899a0713-eb2a-46bb-884d-d9adecdc602e)
![image](https://github.com/metabase/metabase/assets/22608765/cd570657-d997-497c-92f0-afdeebdcca4b)